src/marshaller/HipieDDL.js

```diff
@@ -1394,14 +1399,17 @@
         this.dashboard.marshaller.commsEvent(this, "request", this.request);
         return new Promise(function (resolve, reject) {
             context.comms.call(context.request).then(function (response) {
-                var delay = 500 - (Date.now() - now);  //  500 is to allow for all "clear" transitions to complete...
-                setTimeout(function () {
-                    context.processResponse(response, request, updates).then(function () {
-                        resolve(response);
-                    });
-                    context.dashboard.marshaller.commsEvent(context, "response", context.request, response);
-                    ++context._loadedCount;
-                }, delay > 0 ? delay : 0);
+                var intervalHandle = setInterval(function () {
+                    if (transactionQueue[0] === myTransactionID && Date.now() - now >= 500) {  //  500 is to allow for all "clear" transitions to complete...
```

Comment:
this will probably work on fast computers but on slower computers it may take longer than 1/2 sec.  Is it not possible to have a status object and push/pop holds to it in the transitions and put a loop here waiting until this object says all transition holds are complete?